src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -110,50 +132,52 @@ public static RuntimeGraph Merge(RuntimeGraph left, RuntimeGraph right)
         /// </summary>
         public IEnumerable<string> ExpandRuntime(string runtime)
         {
+            if (IsEmpty)
+            {
+                return new[] { runtime };
+            }
+
             return ExpandRuntimeCached(runtime);
         }
 
         private HashSet<string> ExpandRuntimeCached(string runtime)
         {
-            return _expandCache.GetOrAdd(runtime, r => new HashSet<string>(ExpandRuntimeInternal(r), StringComparer.Ordinal));
-        }
+            return _expandCache!.GetOrAdd(runtime, r => new HashSet<string>(ExpandRuntimeInternal(r), StringComparer.Ordinal));
 
-        /// <summary>
-        /// Expand runtimes in a BFS walk. This ensures that nearest RIDs are returned first.
-        /// Ordering is important for finding the nearest runtime dependency.
-        /// </summary>
-        private IEnumerable<string> ExpandRuntimeInternal(string runtime)
```

Comment:
This method was converted to a local function, as it should only be called on non-empty instances. All callers of `ExpandRuntimeCached` validate the instance is non-empty before calling. The method is otherwise unchanged.